TACC/Core-Portal
# Task/FP-1375: Admin controlled messages

## Overview: ##
Allow arbitrary and dismissable `SectionMessage` generation by admin (through `core/admin`) for components.

## Related Jira tickets: ##

* [FP-1375](https://jira.tacc.utexas.edu/browse/FP-1375)

## Summary of Changes: ##
1. Created a `/intromessages/custom/` endpoint (`GET`: Get templates & messages, `PUT`: Update `unread` state).
2. Create a `CustomMessage` UI component as a wrapper around `SectionMessages` to utilize `state.customMessages`.
3. Implemented necessary sagas/reducers to interact with the `/intromessages/custom/` endpoint.
4. Made messages of type `intro` have a dismiss button.

## Testing Steps: ##
0. Make sure to run migrations within the `core_portal_django` container: `./manage.py migrate`
1. [Make yourself staff or superuser](https://github.com/TACC/Core-Portal/wiki/How-to-Set-Your-User-as-Staff-or-Superuser).
2. Go to `core/admin`.
3. Create a *Custom message template*.
5. Go to the component you made it for (Will need refresh).
6. Ensure that it is correctly rendered
7. Repeat 3-5 for various different configurations (even multiple messages for single component) and ensure that they render accordingly
8. Ensure that dismissed (click `x`) messages do not show up.
9. Go to `core/admin` again.
10. Try removing some that aren't dismissed.
11. Refresh portal.
12. Ensure that the deleted messages do not show.

## UI Photos:
https://user-images.githubusercontent.com/9425579/159995718-6c0688db-797b-4c0a-9105-18c10f1fdca9.mov

## Notes: ##
1. (**For CMD**) Should there be a text limit for messages?
2. I wanted to refactor `intromessages` related components (both frontend/backend) into a more generic `messages` component. But wasn't sure how the migration process would go. Would there be a way to achieve this?
3. (**For CMD**) The `info` message type style does not seem to have an icon (`SectionMessage`, `Message`).  Thus, the dismiss button (`x`), which is a child of the icon does not render.
